js/analyzer.js

```diff
@@ -225,12 +206,9 @@ YoastSEO.Analyzer.prototype.keywordDensityCheck = function() {
  * @returns keywordCount
  */
 YoastSEO.Analyzer.prototype.keywordCount = function() {
-	var keywordMatches = this.preProcessor.__store.cleanTextSomeTags.match( this.keywordRegex );
-	var keywordCount = 0;
-	if ( keywordMatches !== null ) {
-		keywordCount = keywordMatches.length;
-	}
-	this.__store.keywordCount = keywordCount;
+	var matchTextWithWord = require( "./stringProcessing/matchTextWithWord.js" );
+	var keywordCount = matchTextWithWord( this.preProcessor.__store.originalText, this.config.keyword );
```

Comment:
I thought the preprocessor could be removed after these changes, we should at least be able to remove it completely from the analyser.